main.go

```diff
@@ -133,7 +149,7 @@ func doVersion() error {
 	})
 }
 
-func run() int {
+func run(stdin io.Reader) int {
```

Comment:
For testability, I think we should accept `args` and return `result`.

```go
func run(args []string) string
```

Reply:
Happy to make changes here, but I don't quite understand what you want this to look like.  Can you explain more fully?

1. I think you're suggesting to pass in the `args` instead of referring to the global `flag.Args()`.  Are you suggesting parsing the args and etc.. in `main`?
2. I find passing in an io.Reader instead of relying on os.Stdin or mocking it or using real files makes the table driven tests in main_test.go:TestRun cleaner (possible). 
3. What result do you want to return?  It's currently returning success/failure, which is easily testable in `main_test.go:TestRun`, along with the output and stderr.